libsel4/include/interfaces/sel4.xml

```diff
@@ -393,6 +394,54 @@
                 </description>
             </error>
         </method>
+        <method id="TCBSetSchedParams" name="SetSchedParamsFH" manual_name="Set Sched Params (MCS)" manual_label="tcb_setschedparams_mcs">
+            <condition><config var="CONFIG_KERNEL_MCS"/></condition>
+            <brief>
+                Change a thread's priority, maximum controlled priority, scheduling context
+                and fault handler.
+            </brief>
+            <description>
+                <docref>See <autoref label="sec:sched"/></docref>
+            </description>
+            <param dir="in" name="authority" type="seL4_TCB"
+                description="Capability to the TCB to use the MCP from when setting the priority and MCP."/>
+            <param dir="in" name="mcp" type="seL4_Word"
+                description="The thread's new maximum controlled priority."/>
+            <param dir="in" name="priority" type="seL4_Word"
+                description="The thread's new priority."/>
+            <param dir="in" name="sched_context" type="seL4_CPtr"
+                description="Capability to the scheduling context that the TCB should run on. If the scheduling context is already bound to a notification or TCB that is not this TCB this operation will fail. Similarly, if this TCB is already bound to a scheduling context that is not this scheduling context, this will also fail."/>
+            <param dir="in" name="fault_ep" type="seL4_CPtr"
+                description="CPTR to the endpoint which receives IPCs when this thread faults."/>
+            <param dir="in" name="fault_ep_data" type="seL4_Word"
+                description="Optionally set the badge of the new fault handler Endpoint. If set to zero, this parameter has no effect."/>
+            <param dir="in" name="fault_ep_rights" type="seL4_CapRights_t">
+                <description>
+                    Optionally set the rights inherited by the new fault handler Endpoint capability.<docref>Possible values for this type are given in <autoref label="sec:cap_rights"/>  .</docref>
+                    If set to zero, this parameter has no effect.
+                </description>
+            </param>
+            <error name="seL4_IllegalOperation">
+                <description>
+                    The <texttt text="_service"/> is a CPtr to a capability of the wrong type.
+                    Or, <texttt text="_service"/> or <texttt text="sched_context"/> is already bound.
+                    Or, <texttt text="_service"/> is the current thread's TCB.
+                    Or, <texttt text="_service"/> is a TCB in the blocked state and <texttt text="sched_context"/> is not schedulable.
+                </description>
```

Comment:
`updateAndCheckHandlerEP()` can also return `seL4_IllegalOperation now also., when `fault_ep_data` cannot be handled ("handler cap being rebadged when already badged or null")